File: Harish-Kumar/longest-common-prefix.java
/* 

https://leetcode.com/problems/longest-common-prefix/

*/

class Solution {
    public String longestCommonPrefix(String[] strs) {
        String output = strs[0];
        for (int i = 1; i < strs.length; i++) {
            output = compare(output, strs[i]);
        }
        return output;
    }

    public String compare(String output, String str) {
        int n = Math.min(output.length(), str.length());
        int i = 0;
        for (i = 0; i < n; i++) {
            if (output.charAt(i) != str.charAt(i)) {
                break;
            }
        }

        return str.substring(0, i);
    }
}

File: arun_lodhi/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        t= 0
        s = ""
        for x in range(len(min(strs,key = len))):
            count =  0
            for y in strs:
                if strs[0][x]==y[x]:
                    count = count+1
                else:
                    break
            if count==len(strs):
                t = t+1
                count = 0
                continue
            else:
                break
        if t==0:
            return s
        else:
            s = strs[0][0:t]
            return s
                
